float8_experimental/inference.py

```diff
@@ -0,0 +1,257 @@
+        self.quantize_weight()
+
+        if self.activation_casting == ActivationCasting.STATIC:
+            self.register_buffer("activation_scale", quant_config.activation_scale)
+        else:
+            self.activation_scale = None
+        return self
+
+    @classmethod
+    def from_float(
+        cls,
+        module: nn.Module,
+        quant_config: QuantConfig,
+    ) -> "Float8LinearInference":
+        """
+        Create an nn.Linear with fp8 compute from a regular nn.Linear
+
+        Args:
+            mod (torch.nn.Linear): nn.Linear to convert
+            quant_config (QuantConfig): Configuration for the weight and activation casting
+        """
+        return (
+            cls.create_meta_class(module.in_features, module.out_features)
+            .set_weight_and_bias(module.weight, module.bias)
+            .set_mm_config()
+            .set_quantization_config(quant_config)
+        )
+
+
+def cast_to_float8_e4m3fn(
+    inpt_tensor: torch.Tensor,
+    mm_config: ScaledMMConfig,
+    reduce_amax: bool = False,
+    activation_scale: Optional[torch.Tensor] = None,
+) -> Float8Tensor:
+    """Casts an input tensor to the Float8 (e4m3fn) format for efficient computation.
+
+    Args:
+        inpt_tensor: The input tensor to be cast.
+        mm_config: Configuration settings for the matrix multiplication
+        reduce_amax: Whether to reduce the amax (absolute maximum) among the local distributed group.
+        activation_scale: Optional tensor specifying the scale for activation. Default is None.
+
+    Returns:
+        Float8Tensor: The input tensor cast to Float8 (e4m3fn) format.
+
+    Note:
+        If the input tensor is already in Float8 format, it is returned as is without re-casting.
+    """
+    if tensor_already_casted_to_fp8(inpt_tensor):
+        return inpt_tensor
+    scale = (
+        activation_scale
+        if activation_scale is not None
+        else tensor_to_scale(inpt_tensor, e4m3_dtype, reduce_amax)
+    )
+    return Float8Tensor.to_float8(inpt_tensor, scale, e4m3_dtype, mm_config=mm_config)
+
+
+def quantize_to_float8(
+    module: nn.Module,
+    quant_config: QuantConfig,
+    *,
+    skip_fqn_list: Optional[List[str]] = None,
+    linear_layer_filter: Optional[Callable[[nn.Linear], bool]] = None,
+) -> nn.Module:
+    """
+    Converts torch.nn.Linear layers in the given module to Float8LinearInference.
+
+    Note:
+        If applied to a root-level nn.Linear, the module will not be modified in place
+        and returned instead
+
+    Args:
+        module (nn.Module): The module to modify.
+        quant_config (QuantConfig): Quantization configuration for Float8 conversion.
+        skip_fqn_list (List[str], optional): List of module FQNs to skip during conversion.
+        linear_layer_filter (Callable[[nn.Linear], bool], optional): Function to filter
+            which Linear layers should be converted.
```

Comment:
this should be inified (in the training code too), maybe add a comment mentioning that?